app/src/main/java/appricottsoftware/clarity/LoginActivity.java

```diff
@@ -3,44 +3,279 @@
 import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
+import com.facebook.AccessToken;
+import com.facebook.AccessTokenTracker;
+import com.facebook.GraphRequest;
+import com.facebook.GraphResponse;
+import com.facebook.LoggingBehavior;
+import com.facebook.Profile;
+import com.facebook.login.widget.LoginButton;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.SignInButton;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.Task;
+import com.loopj.android.http.JsonHttpResponseHandler;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import appricottsoftware.clarity.sync.ClarityApp;
+
+import java.util.Arrays;
+
+import appricottsoftware.clarity.sync.ClarityClient;
 import butterknife.BindView;
 import butterknife.ButterKnife;
+import cz.msebera.android.httpclient.Header;
+
+import com.facebook.CallbackManager;
+import com.facebook.FacebookCallback;
+import com.facebook.FacebookException;
+import com.facebook.FacebookSdk;
+import com.facebook.login.LoginResult;
 
 public class LoginActivity extends AppCompatActivity implements View.OnClickListener {
 
-    @BindView(R.id.bt_login) Button btLogin;
-    @BindView(R.id.bt_register) Button btRegister;
+    @BindView(R.id.btn_login) Button btnLogin;
+    @BindView(R.id.btn_register) Button btnRegister;
+    @BindView(R.id.btn_loginFacebook) LoginButton btnFacebook;
+    @BindView(R.id.btn_loginGoogle) SignInButton btnGoogle;
+    @BindView(R.id.et_email) EditText etEmail;
+    @BindView(R.id.et_password) EditText etPassword;
+
+    private static final String EMAIL = "email";
+    private CallbackManager fbCallbackManager;
+    private AccessTokenTracker fbAccessTokenTracker;
+    private GoogleSignInAccount googleAccount;
+    private GoogleSignInClient googleSignInClient;
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
         ButterKnife.bind(this);
 
-        btLogin.setOnClickListener(this);
-        btRegister.setOnClickListener(this);
+        btnLogin.setOnClickListener(this);
+        btnRegister.setOnClickListener(this);
+        btnGoogle.setOnClickListener(this);
+        btnGoogle.setSize(SignInButton.SIZE_WIDE);
+
+        facebookLogin();
+        googleLogin();
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        googleAccount = GoogleSignIn.getLastSignedInAccount(this);
```

Comment:
It's handled within the facebookLogin() since it has it's own object handling most of the login process. From the tutorials and facebook website there doesn't seem to be an equivalent getLastSignedInAccount. Currently, within facebookLogin() I refer to access tokens and profiles to infer if they're loggin in or not.

Reply:
I saw it was handled in the facebookLogin() function, but I think it would make more sense to make a method to try to auto sign in to facebook and put it in the same place. It is confusing to have the same functionality placed in multiple places.

Reply:
I think I found a decent compromise that still lets the session persist even after closing the app